sw/device/lib/dif/dif_pwrmgr.c

```diff
@@ -4,8 +4,543 @@
+ * definitions in the header, which we rely on for a simpler implementation.
+ * These constants and their usages must be revisited if there is a change in
+ * hardware.
+ */
+
+/**
+ * Relevant bits of the control register must start at
+ * `PWRMGR_CONTROL_CORE_CLK_EN` and be in the same order as
+ * `dif_pwrmgr_domain_option_t` constants.
+ */
+_Static_assert(kDifPwrmgrDomainOptionCoreClockInLowPower ==
+                   (1u << (PWRMGR_CONTROL_CORE_CLK_EN -
+                           PWRMGR_CONTROL_CORE_CLK_EN)),
+               "Layout of control register changed.");
+
+_Static_assert(kDifPwrmgrDomainOptionIoClockInLowPower ==
+                   (1u << (PWRMGR_CONTROL_IO_CLK_EN -
+                           PWRMGR_CONTROL_CORE_CLK_EN)),
+               "Layout of control register changed.");
+
+_Static_assert(kDifPwrmgrDomainOptionUsbClockInLowPower ==
+                   (1u << (PWRMGR_CONTROL_USB_CLK_EN_LP -
+                           PWRMGR_CONTROL_CORE_CLK_EN)),
+               "Layout of control register changed.");
+
+_Static_assert(kDifPwrmgrDomainOptionUsbClockInActivePower ==
+                   (1u << (PWRMGR_CONTROL_USB_CLK_EN_ACTIVE -
+                           PWRMGR_CONTROL_CORE_CLK_EN)),
+               "Layout of control register changed.");
+
+_Static_assert(kDifPwrmgrDomainOptionMainPowerInLowPower ==
+                   (1u << (PWRMGR_CONTROL_MAIN_PD_N -
+                           PWRMGR_CONTROL_CORE_CLK_EN)),
+               "Layout of control register changed.");
+
+/**
+ * Bitfield for interpreting the configuration options in the control
+ * register.
+ */
+static const bitfield_field32_t kDomainConfigBitfield = {
+    .mask = kDifPwrmgrDomainOptionCoreClockInLowPower |
+            kDifPwrmgrDomainOptionIoClockInLowPower |
+            kDifPwrmgrDomainOptionUsbClockInLowPower |
+            kDifPwrmgrDomainOptionUsbClockInActivePower |
+            kDifPwrmgrDomainOptionMainPowerInLowPower,
+    .index = PWRMGR_CONTROL_CORE_CLK_EN,
+};
+
+/**
+ * Relevant bits of the WAKEUP_EN and WAKE_INFO registers must start at `0` and
+ * be in the same order as `dif_pwrmgr_wakeup_request_source_t` constants.
+ */
+_Static_assert(kDifPwrmgrWakeupRequestSourceOne ==
+                   (1u << PWRMGR_WAKEUP_EN_EN_0),
+               "Layout of WAKEUP_EN register changed.");
+_Static_assert(kDifPwrmgrWakeupRequestSourceOne ==
+                   (1u << PWRMGR_WAKE_INFO_REASONS),
+               "Layout of WAKE_INFO register changed.");
+
+/**
+ * Relevant bits of the RESET_EN register must start at `0` and be in the same
+ * order as `dif_pwrmgr_reset_request_source_t` constants.
+ */
+_Static_assert(kDifPwrmgrResetRequestSourceOne == (1u << PWRMGR_RESET_EN_EN_0),
+               "Layout of RESET_EN register changed.");
+
+/**
+ * `dif_pwrmgr_irq_t` constants must match the corresponding generated values.
+ */
+_Static_assert(kDifPwrmgrIrqWakeup == PWRMGR_INTR_COMMON_WAKEUP,
```

Comment:
Why not just use an auxilliary switch statement helper function like other DIFs do?

This would avoid introducing unnecessary additional constants.

Reply:
A switch would use the same constants. Defining enums to match those values eliminates the need for an additional function.